[IRGen] Don't compute witness table layout just to form existentials

jrose-apple opened the pull request:
We weren't using this information, and trying to access it runs into trouble when the protocol is in another file.

jrose-apple committed:
[IRGen] Remove unused ExistentialTypeInfoBase::findWitnessTable

Turns out we never ask this of TypeInfo anymore.

jrose-apple commented:
@swift-ci Please test

jrose-apple commented:
@swift-ci Please test source compatibility

jrose-apple commented:
@swift-ci Please smoke test compiler performance

rjmccall commented on lib/IRGen/GenExistential.cpp:
```diff
@@ -162,49 +163,11 @@ namespace {
-        [&](unsigned originIndex) {
-          return witnesses[originIndex];
-        });
-    }
```
Was this all dead?  Unfortunate.

rjmccall reviewed (approved):
I assume the problem is that building an existential type alone doesn't validate the protocol enough to perform a layout of it?

Anyway, LGTM.

jrose-apple commented:
Yep, that's correct. And yeah, I don't know if the dead code was *never* used or only *recently* unused, but that was the only code that wanted the ProtocolInfo.

slavapestov commented:
Nice fix!

jrose-apple committed:
[IRGen] Don't compute witness table layout just to form existentials

We weren't using this information, and trying to access it runs into
trouble when the protocol is in another file.

jrose-apple commented:
Restarting tests due to fixing 80-column violations.

[Compiler perf](
https://ci.swift.org/job/swift-PR-compiler-performance-smoke-test-macOS/313/)
[Source compat](
https://ci.swift.org/job/swift-PR-source-compat-suite/1672/)
[Linux full](
https://ci.swift.org/job/swift-PR-Linux/6778/)
[macOS full](
https://ci.swift.org/job/swift-PR-osx/6772/)

@swift-ci Please smoke test
